File: packages/core/src/memory/MemoryEngine.ts
import { MemoryItem, createMemoryItem } from "./MemoryItem";
import { MemoryStore } from "./MemoryStore";

export class MemoryEngine {
  constructor(private store: MemoryStore) {}

  async addMemory(
    userId: string,
    content: string,
    tags: string[] = [],
    importance: number = 0.5
  ): Promise<MemoryItem> {
    const memory = createMemoryItem(userId, content, tags, importance);
    await this.store.add(memory);
    return memory;
  }

  async retrieveMemories(
    userId: string,
    query: string,
    limit?: number
  ): Promise<MemoryItem[]> {
    return await this.store.retrieve(userId, query, limit);
  }

  async listMemories(userId: string): Promise<MemoryItem[]> {
    return await this.store.list(userId);
  }

  async clearMemories(userId: string): Promise<void> {
    await this.store.clear(userId);
  }
}

export function createMemoryEngine(store: MemoryStore): MemoryEngine {
  return new MemoryEngine(store);
}

File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/memory/MemoryEngine.kt
package ai.mobileagentos.compose.memory

class MemoryEngine(private val store: MemoryStore) {
    
    suspend fun addMemory(
        userId: String,
        content: String,
        tags: List<String> = emptyList(),
        importance: Float = 0.5f
    ): MemoryItem {
        val memory = MemoryItem.create(userId, content, tags, importance)
        store.add(memory)
        return memory
    }
    
    suspend fun retrieveMemories(
        userId: String,
        query: String,
        limit: Int = 5
    ): List<MemoryItem> {
        return store.retrieve(userId, query, limit)
    }
    
    suspend fun listMemories(userId: String): List<MemoryItem> {
        return store.list(userId)
    }
    
    suspend fun clearMemories(userId: String) {
        store.clear(userId)
    }
}